crates/astria-sequencer-utils/src/blob_parser.rs

```diff
@@ -0,0 +1,797 @@
+use std::{
+    fmt::{
+        self,
+        Display,
+        Formatter,
+        Write,
+    },
+    num::NonZeroUsize,
+};
+
+use astria_core::{
+    brotli::decompress_bytes,
+    generated::sequencerblock::v1alpha1::{
+        rollup_data::Value as RawRollupDataValue,
+        Deposit as RawDeposit,
+        RollupData as RawRollupData,
+        SubmittedMetadata as RawSubmittedMetadata,
+        SubmittedMetadataList as RawSubmittedMetadataList,
+        SubmittedRollupData as RawSubmittedRollupData,
+        SubmittedRollupDataList as RawSubmittedRollupDataList,
+    },
+    primitive::v1::RollupId,
+    sequencerblock::v1alpha1::{
+        block::{
+            Deposit,
+            DepositError,
+            SequencerBlockHeader,
+        },
+        celestia::{
+            SubmittedRollupData,
+            UncheckedSubmittedMetadata,
+            UncheckedSubmittedRollupData,
+        },
+    },
+};
+use astria_eyre::eyre::{
+    bail,
+    Result,
+    WrapErr,
+};
+use astria_merkle::audit::Proof;
+use base64::{
+    prelude::BASE64_STANDARD,
+    Engine,
+};
+use clap::ValueEnum;
+use colour::write_blue;
+use ethers_core::types::{
+    transaction::eip2930::AccessListItem,
+    Transaction,
+};
+use indenter::indented;
+use itertools::Itertools;
+use prost::{
+    bytes::Bytes,
+    Message,
+};
+use serde::Serialize;
+
+#[derive(clap::Args, Debug)]
+pub struct Args {
+    /// Base64-encoded blob data
+    #[arg(value_name = "BLOB")]
+    input: String,
```

Comment:
is the blob just pasted in directly to this command? should there be an option for reading from a file perhaps?

Reply:
Good call - I'll add an option to get the input from a file too.

Reply:
To pile onto this, I think it would be nice if this could also read input from STDIN, maybe after a `--` flag.

Reply:
Done in [f16e647](https://github.com/astriaorg/astria/pull/1059/commits/f16e647ec145aa761ed0c4b8a70227911a9d1fe6).